synapse/handlers/auth.py

```diff
@@ -885,6 +889,18 @@ async def _find_user_id_and_pwd_hash(
             )
         return result
 
+    def get_allow_change_password(self) -> bool:
+        """Get status if users on this server are allowed to change or set a password.
+        Both `config.password_enabled` and `config.password_localdb_enabled`
+        must be true.
+        It is allowed that people add a password to their account even if they created
+        that account with SSO if it is not explicitly prohibited.
+
+        Returns:
+            Whether users on this server are allowed to change or set a password
+        """
+        return self._change_password
```

Comment:
I don't think we need the `_change_password` property here, we can return `self._password_enabled and self._password_localdb_enabled`.